controllers/cbcontainerscluster_controller_test.go

```diff
@@ -173,12 +194,56 @@ func TestClusterReconcile(t *testing.T) {
 	})
 
 	t.Run("When state applier returns state was not changed, reconcile should return default Requeue", func(t *testing.T) {
-		result, err := testCBContainersClusterController(t, setupClusterCustomResource, setUpTokenSecretValues, func(testMocks *ClusterControllerTestMocks) {
+		result, err := testCBContainersClusterController(t, setupClusterCustomResource, setUpTokenSecretValues, setUpGateway, setUpOperatorVersionProvider, func(testMocks *ClusterControllerTestMocks) {
 			testMocks.ClusterProcessor.EXPECT().Process(&ClusterCustomResourceItems[0], MyClusterTokenValue).Return(secretValues, nil)
 			testMocks.ClusterStateApplier.EXPECT().ApplyDesiredState(testMocks.ctx, &ClusterCustomResourceItems[0], secretValues, testMocks.client, gomock.Any()).Return(false, nil)
 		})
 
 		require.NoError(t, err)
 		require.Equal(t, result, ctrlRuntime.Result{})
 	})
+
+	t.Run("When it gets the compatibility matrix succesfully it should make a compatibility check", func(t *testing.T) {
+		t.Run("When agent version is between min and max it should not return an error", func(t *testing.T) {
+			result, err := testCBContainersClusterController(t, setupClusterCustomResource, setUpTokenSecretValues, func(testMocks *ClusterControllerTestMocks) {
+				testMocks.gatewayCreator.EXPECT().CreateGateway(gomock.Any(), gomock.Any()).Return(testMocks.gateway)
+
+				operatorVersion := "3.1.0"
+				testMocks.operatorVersionProvider.EXPECT().GetOperatorVersion().Return(operatorVersion, nil)
```

Comment:
[Minor] You could use the value provided by `setupOperatorVersionProvider` and utilize `gomock.Any()` when setting up the compatibility matrix mock? Since the operator version is essentially meaningless to this test, the agent version is what matters.

Reply:
I believe that is wrong. `GetCompatibilityMatrixEntryFor` should be called with whatever comes from `operatorVersionProvider.GetOperatorVersion`. the test currently verifies that. if I use `gomock.Any()` it will not verify that and it will be less reliable